core/src/main/java/org/elasticsearch/common/path/PathTrie.java

```diff
@@ -26,6 +26,28 @@
 import static java.util.Collections.unmodifiableMap;
 
 public class PathTrie<T> {
+    
+    public enum TrieMatchingMode {
```

Comment:
it would we nice if we can make this an impl detail and just offer a method that returns a list or a stream of `<T>` instead of exposing this explicitly.